cockatrice/src/deck_loader.cpp

```diff
@@ -108,3 +110,72 @@ DeckLoader::FileFormat DeckLoader::getFormatFromName(const QString &fileName)
     }
     return PlainTextFormat;
 }
+
+bool DeckLoader::saveToStream_Plain(QTextStream &out)
+{
+    // deck header
+    if(!getName().isEmpty())
+        out << "// " << getName() << "\n\n";
+    if(!getComments().isEmpty())
+    {
+        QStringList commentRows = getComments().split(QRegExp("\n|\r\n|\r"));
+        foreach(QString row, commentRows)
+            out << "// " << row << "\n";
+        out << "\n";
+    }
+
+    // loop zones
+    for (int i = 0; i < getRoot()->size(); i++) {
+        const InnerDecklistNode *node =
+            dynamic_cast<InnerDecklistNode *>(getRoot()->at(i));
+
+        // group cards by card type and count the subtotals
+        QMultiMap<QString, DecklistCardNode*> cardsByType;
+        QMap<QString, int> cardTotalByType;
+        int cardTotal = 0;
+
+        for (int j = 0; j < node->size(); j++) {
+            DecklistCardNode *card =
+                dynamic_cast<DecklistCardNode *>(
+                    node->at(j)
+                );
+
+            CardInfo *info = db->getCard(card->getName());
+            QString cardType = info ? info->getMainCardType() : "unknown";
+
+            cardsByType.insert(cardType, card);
+
+            if(cardTotalByType.contains(cardType))
+                cardTotalByType[cardType] += card->getNumber();
+            else
+                cardTotalByType[cardType] = card->getNumber();
+
+            cardTotal += card->getNumber();
+        }
+
+        out << "// " << cardTotal << " " << node->getVisibleName() << "\n";
+
+        // print cards to stream
+        foreach(QString cardType, cardsByType.uniqueKeys())
```

Comment:
IMO it's easier to test/edit the code if we split side effects from computation. I'd prefer splitting this into two functions. One that just processes the deck into the structure needed for export and another that exports the text. We could then reuse the processing function for different styles of export - plain text, tappedout, deckstats, test code, etc.